ci-templates/gitlab/image-builder.yml

```
@@ -420,3 +423,24 @@ t4c/client/ease/remote/debug:
       --build-arg BASE_IMAGE=$BASE_IMAGE \
       ease/debug
     - *push
+
+jupyter:
+  stage: build
+  needs:
+    - job: base
+      optional: true
+  rules:
+    - if: '$JUPYTER == "1"'
+      when: always
+  variables:
+    BASE_IMAGE: $DOCKER_REGISTRY/base2
+    IMAGE: $DOCKER_REGISTRY/jupyter-notebook
```

```suggestion
  rules:
    - if: '$JUPYTER == "1"'
      when: always
  before_script:
    - export DOCKER_TAG=$JUPYTER_VERSION-$(echo $CAPELLA_DOCKER_IMAGES_REVISION | sed 's/[^a-zA-Z0-9.]/-/g')-$CI_COMMIT_REF_NAME    
  variables:
    BASE_IMAGE: $DOCKER_REGISTRY/base2
    IMAGE: $DOCKER_REGISTRY/jupyter-notebook
```
This is needed for the comment below and in addition, we would need to add a `JUPYTER_VERSION` variable. But I think it makes sense to already do it like this so we already minimize potential changes if we want to, for example have notebook images for python 3.11 and an experimental one for 3.12.